### ISSUE TITLE
Keyboard focus when opening file dialog

### ISSUE THREAD

**Issue by zunino** [brave/brave-browser#30856]
## Description 
When opening a file dialog (CTRL+O), keyboard focus is not moved to the new window. That is non-intuitive and breaks the workflow. Whenever I use the keyboard shortcut to bring up the file dialog, I'm forced to reach for my pointing device in order to set focus on, say, the file filter field so I can input data. Ideally, users should be able to stick to keyboard commands (e.g. CTRL+L, TAB, arrows) to interact with the dialog.

## Steps to Reproduce
   1. Press CTRL+O.
   2. Press the down arrow key one or more times.

## Actual result:
The keyboard input is handled by the browser window in the background and not the file dialog.

## Expected result:
Keyboard focus should be set on the (modal) dialog that has just been opened so that the interaction flow is not disrupted.

## Reproduces how often: 
Easily reproduced.

## Brave version (brave://version info)
Brave: 1.52.117 Chromium: 114.0.5735.90 (Official Build) (64-bit)
Revision: 386bc09e8f4f2e025eddae123f36f6263096ae49-refs/branch-heads/5735@{#1052}
OS: Linux

## Version/Channel Information:
- Can you reproduce this issue with the current release? YES
- Can you reproduce this issue with the beta channel? N/A
- Can you reproduce this issue with the nightly channel? N/A

## Other Additional Information:
- Does the issue resolve itself when disabling Brave Shields? NO
- Does the issue resolve itself when disabling Brave Rewards? NO
- Is the issue reproducible on the latest version of Chrome? I've just tested it on Chromium Version 113.0.5672.126 (Official Build) for Linux Mint (64-bit) and the behavior also manifests there, which seems to indicate Brave has merely inherited it.

## Miscellaneous Information:
As mentioned previously, the current behavior breaks the flow and hurts accessibility (keyboard focus management is crucial for, e.g. users with impaired vision or motor abilities).
